validator/src/main.rs

```diff
@@ -576,8 +576,16 @@ pub fn main() {
             Arg::with_name("minimal_rpc_api")
                 .long("--minimal-rpc-api")
                 .takes_value(false)
+                .hidden(true)
                 .help("Only expose the RPC methods required to serve snapshots to other nodes"),
         )
+        .arg(
+            Arg::with_name("extended_rpc_api")
```

Comment:
Code looks great, I just want to bikeshed the name a little.  I feel `--full-rpc-api` is more accurate.  Alternatively `--complete-rpc-api`?

Reply:
I also prefer something like this ^
I've no preference between "full" and "complete"

Reply:
Yeah I was wrestling with whether we should commit to this set being "full/complete" given the deprecated method set.

I'd additionally considered something like `--enable-rpc-method-set`, which takes a CSV following the available traits `[minimal*, bank, accounts, deprecated-v1.9, deprecated-v1.7, full, obsolete-v17]`? Then we additionally hide the `--enable-rpc-obsolete_v1_7` flag

Reply:
`--full-rpc-api` feels like a nice complement to `--minimal-rpc-api` to me.

I'm not opposed to something like `--rpc-method-set` but also not sure it buys us too much either. The common options seem to be "minimal", "full", or "full" with "enable-rpc-obsolete_v1_7".  It's unclear to me who would want more granular control over the exposed methods

Reply:
> but also not sure it buys us too much either

Well surely we have the additional complexity in the code base for a reason, right? :wink: 

I'll go with `--full-rpc-api` from now